### PR TITLE
[REG-533] Python timer support and build warning fixes

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- TODO: Publish RGBot and update versions in GameMachine/RG-CTF-Utils/System-Bots/etc

### CODE REVIEW

File: lib/RGBot.js
Comment (abeizer) [inline lib/RGBot.js:155]:
```suggestion
     * @param {Bot} bot Mineflayer bot instance
```

Bot isn't optional

File: lib/RGBot.js
Comment (abeizer) [inline lib/RGBot.js:530]: shouldn't this use the setTimout  method you defined below so there's no change to this method?

File: lib/RGBot.js
Comment (abeizer) [inline lib/RGBot.js:546]: Looks like action is required here, unless we default it to `() => {}`.
If we want to make it optional, suggest reordering params so that time is first

File: lib/RGBot.js
Comment (abeizer) [inline lib/RGBot.js:753]: While we're here, `pathFunc` isn't optional

File: lib/RGBot.js
Comment (abeizer) [inline lib/RGBot.js:752]: This is misleading because it makes it sound like I'll still perform the movement, I just won't have stuck detection but this just immediately returns `!stuck` if I'm python

**Review by abeizer — CHANGES_REQUESTED** [review]

File: lib/RGBot.js
Comment (RG-nAmKcAz) [reply lib/RGBot.js:158]: good suggestion, will make this into options object

File: lib/RGBot.js
Comment (RG-nAmKcAz) [reply lib/RGBot.js:546]: was intended to be required.. fixing jsdoc

File: lib/RGBot.js
Comment (RG-nAmKcAz) [reply lib/RGBot.js:530]: changing for consistency

File: lib/RGBot.js
Comment (vontell) [inline lib/RGBot.js:1331]: Wouldn't this never resolve? Should it be this? (not actually sure myself)
```typescript
return new Promise( resolve => resolve(this.#bot.heldItem));
```

File: package.json
Comment (vontell) [inline package.json:3]: You should be able to leave this at whatever I have it in the newest package json, since I haven't published yet.

**Review by vontell — APPROVED** [review]
Approving assuming Abby's comments and my comments are addressed! They were small things so figured I'd approve

File: package.json
Comment (RG-nAmKcAz) [reply package.json:3]: ok this time, but i don't agree with having to trust/know that you won't publish something that's already merged before mine gets published/merged.. that seems like we're playing with fire

File: lib/RGBot.js
Comment (RG-nAmKcAz) [reply lib/RGBot.js:1331]: yep, my mistake.. fixing

File: package.json
Comment (RG-nAmKcAz) [reply package.json:3]: actually, I added apis ,shouldn't this be a minor update after all ?

File: lib/RGBot.js
Comment (RG-nAmKcAz) [reply lib/RGBot.js]: corrected

File: lib/RGBot.js
Comment (RG-nAmKcAz) [reply lib/RGBot.js:52]: oddly, the only conflict was in package.json.. anything specific I should be checking for in the js code ?

Comment (vontell) [reply lib/RGBot.js:52]: Oh yes - note that this line appears twice, as I have it added on line 58 below

File: package.json
Comment (vontell) [reply package.json:3]: I think we need to put together some docs on how we version... for instance, for rg-service, we don't update the version on every commit, only every release...

Happy with this being a minor update and publishing for now.

**Review by vontell — APPROVED** [review]

**Review by vontell — CHANGES_REQUESTED** [review]
(Changing to request change due to pathfinder comment)

**Comment by RG-nAmKcAz** [comment]
@vontell Fixed pathfinder line

Also, we version this on each update because we release it on each update when we publish to npm.  Thus to me the version updates seem appropriate

**Comment by vontell** [comment]
> Also, we version this on each update because we release it on each update when we publish to npm. Thus to me the version updates seem appropriate

Yeah I think that makes sense and is fine - I personally have not published on each update, which I think was the difference... so for now happy to keep incrementing the version number!

**Review by vontell — APPROVED** [review]

**Review by abeizer — APPROVED** [review]

**Pull request merged**
